functions/construction/test_input_data.R

```diff
@@ -0,0 +1,73 @@
+#' Test .csv files created by metric leads
+#'
+#' @param data A data frame
+#' @param geography "county" or "place"
+#' @param subgroups A character vector of subgroups expected in the subgroup file
+#'
+#' @return The function will fail if the data do not meet certain criteria. 
+#' Returns a tibble with the number of geoids per year.
+#'
+test_input_data <- function(data, geography = "county", subgroups = NULL, confidence_intervals = TRUE) {
+  
+  # check that the file has the necessary columns
+  # check that the first few columns are year, state
+  data_names <- names(data)
+  
+  stopifnot(data_names[1] == "year")
+  stopifnot(data_names[2] == "state")
+  stopifnot(data_names[3] == "county" | data_names[3] == "place")
+  
+  if (!is.null(subgroups)) {
+    
+    stopifnot(data_names[4] == "subgroup_type")
+    stopifnot(data_names[5] == "subgroup")
+    
+  }
+
+  #Check that NA values for confidence intervals and quality variables align 
+  if (isTRUE(confidence_intervals)) {
+    
+    stopifnot(sum(is.na(select(data, ends_with("lb")))) == sum(is.na(select(data, ends_with("quality")))))
```

Comment:
Formatting! This is so tough to reach. 

Always refer to "_lb" instead of "lb" and "_quality" instead of "quality".

* compare quality and lb
* compare quality and ub
* compare lb and ub